### PR TITLE
[#9551] Standardize modals

### LINKED ISSUE #9551: Instructor: Correct wrong colors and ensure consistency between modal boxes
**Environment**:
`master` branch

**Steps to reproduce/Expected behavior**
The modal boxes present in the instructor pages are inconsistent and some have wrong colors. Below are a few examples. There might be more.
1. `/web/instructor/home`
- `Course` -> `Archive` 
image
- `Course` -> Header's `Delete` & Row's `Delete`
image
image
2. `/web/instructor/courses`
- `Course`'s row -> `Delete`
image
3. `/web/instructor/courses/details?courseid=inst.exa-demo`
- Row's `delete (`student-list-component`)
image

Also, some modal boxes have excess padding and should be removed. Example below:
image

**Expected behavior**
1. Warning modal boxes should be using the same yellow hex code
2. Delete modal boxes should be using the same red hex code
3. Archive/normal message modal boxes should be using the same blue hex code

### PR DESCRIPTION
**Pull request by ccyccyccy**
Fixes #9551 
Continuation / Adaptation of #9843 

~Need some reviews on the approach of calling the modal. (`ConfirmationModalModule`)~

**Components refactored**

- question-edit-form.component
- instructor-home-page.component
- Student-profile-page.component
- contribution-question-edit-answer-form.component
- Contribution-question-statistics.component
- More-info.component
- instructor-course-enroll-page.component
- Instructor-help-questions-section.component
- Student-list.component
- instructor-session-edit-page.component
- instructor-course-student-edit-page.component
- Instructor-courses-page.component
- Instructor-course-edit-page.component
- Session-submission-page.component
- Admin-search-page.component
- Rubric-question-edit-details-form.component
- Session-edit-form.component

**Modals left alone** (more complex modals)
- Copy-from-other-sessions-modal.component
- Copy-questions-from-other-sessions-modal.component
- Copy-session-modal.component
- #resendPastLinksModal
- Resend-results-link-to-students-modal.component
- Saving-complete-modal.component
- Send-reminders-to-student-modal.component
- Template-question-modal.component
- Upload-edit-profile-picture-modal.component
- View-role-privileges-modal.component
- Sessions-permanent-deletion-confirm-modal.component
- Comment-table-modal-component

For modals left alone, I plan to standarise their style after this PR is merged.

**Deleted modal components**

- confirm-delete-comment-modal.component
- Course-permanent-deletion-confirm-modal.component
- Course-soft-deletion-confirm-modal.component
- Delete-instructor-confirm-modal.component
- Feedback-session-closed-modal.component
- Feedback-session-closing-soon-modal.component
- Feedback-session-deleted-modal.component
- Feedback-session-not-open-modal.component 
- Regenerate-links-confirm-modal.component
- Resend-invitation-email-modal.component
- Status-message-modal.component
- Reset-google-id-confirm-modal.component
- Confirm-publishing-session-modal.component
- Confirm-unpublishing-session-modal.component
- confirm-session-move-to-recycle-bin-modal.component
- Confirm-session-move-to-recycle-bin-modal.component

**Modal design:**
<pre>
INFO   | WARNING
DANGER | NEUTRAL
</pre>

image

### CODE REVIEW

File: src/web/services/confirmation-modal.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmationModalType } from '../app/components/confirmation-modal/confirmation-modal-type';
+import { ConfirmationModalComponent } from '../app/components/confirmation-modal/confirmation-modal.component';
+
+/**
+ * Service to handle display of confirmation modals.
+ * For more a complicated use case, use {@link NgbModal} directly.
```
Comment (wkurniawan07) [inline src/web/services/confirmation-modal.service.ts:8]: What kind of "more complicated use case" do we have right now?

Comment (ccyccyccy) [reply src/web/services/confirmation-modal.service.ts:8]: Modals like `CopyQuestionsFromOtherSessionsModalComponent` and `TemplateQuestionModalComponent` would be hard to fit into the standardized modal. i.e. modals that has non text only contents that involves callbacks and have local states.

Comment (xpdavid) [reply src/web/services/confirmation-modal.service.ts:8]: I think you can name the method as `openSimpleModel`.

File: src/web/services/confirmation-modal.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  /**
+   * Opens a confirmation modal
+   * @param header to be displayed on the modal
+   * @param type which determines the look of the modal
+   * @param content to be displayed in the body of the modal. content supports HTML tags
+   * @param options See {@code ConfirmationModalOptions}
+   */
+  open(header: string, type: ConfirmationModalType, content: any, options?: ConfirmationModalOptions): NgbModalRef {
```
Comment (xpdavid) [inline src/web/services/confirmation-modal.service.ts:25]: Can we give `string` type for content?

File: src/web/services/confirmation-modal.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return modalRef;
+  }
+}
+
+/**
+ * isNotificationOnly: true will cause modal to only have 1 button
+ * confirmMessage: custom text message for confirm button
+ * cancelMessage: custom text message for cancel button
+ */
+export interface ConfirmationModalOptions {
```
Comment (xpdavid) [inline src/web/services/confirmation-modal.service.ts:44]: Better to put definition on the top.

File: src/web/services/confirmation-modal.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      });
+    }
+    return modalRef;
+  }
+}
+
+/**
+ * isNotificationOnly: true will cause modal to only have 1 button
+ * confirmMessage: custom text message for confirm button
+ * cancelMessage: custom text message for cancel button
```
Comment (xpdavid) [inline src/web/services/confirmation-modal.service.ts:42]: The explanation for each field should be put along with field itself.

File: src/web/app/components/comment-box/comment-row/comment-row.component.ts [hunk 5]
```diff
@@ -114,7 +116,8 @@ export class CommentRowComponent implements OnInit, OnChanges {
 
   visibilityStateMachine: CommentVisibilityStateMachine;
 
-  constructor(private modalService: NgbModal, private commentService: FeedbackResponseCommentService) {
+  constructor(private confirmationModalComponent: ConfirmationModalService,
```
Comment (xpdavid) [inline src/web/app/components/comment-box/comment-row/comment-row.component.ts:119]:
```suggestion
  constructor(private confirmationModalService: ConfirmationModalService,
```

File: src/web/services/confirmation-modal.service.ts (hunk 1 above)
Comment (wkurniawan07) [reply src/web/services/confirmation-modal.service.ts:8]: I see, that's right. Our target here is for simple enough modals, e.g. those that require no action or "yes/no" type of action.

Comment (ccyccyccy) [reply src/web/services/confirmation-modal.service.ts:8]: I renamed `confirmation-modal.service` to `simple-modal.service`, and then have 2 different methods `openConfirmationModal` and `openInformationModal` for modals that requires confirmation or no action respectively.

**Comment by ccyccyccy** [comment]
@damithc 

There are mainly 2 styles of modals used throughout the website. Which ones do you think is better?
image

**Comment by damithc** [comment]
> There are mainly 2 styles of modals used throughout the website. Which ones do you think is better?
> image

I'm OK with either. You guys choose.
As per screenshot, the phrasing of the modal heading needs to be standardized too. E.g., always phrase like a question for which the buttons are the answers.

File: src/web/app/pages-help/instructor-help-page/instructor-help-questions-section/instructor-help-questions-section.component.ts [hunk 6]
```diff
@@ -127,17 +128,186 @@ export class InstructorHelpQuestionsSectionComponent extends InstructorHelpSecti
...
 
   /**
-   * Opens modal window.
+   * Opens modal for contribution info.
    */
-  openModal(modal: any): void {
-    this.modalService.open(modal);
+  openContribInfoModal(): void {
+    const modalContent: string = `<div class="modal-body">
+      <ul>
```
Comment (wkurniawan07) [inline src/web/app/pages-help/instructor-help-page/instructor-help-questions-section/instructor-help-questions-section.component.ts:142]: We're putting too much of HTML here. If there is this large amount of HTML code necessary, should keep the HTML code inside the HTML file and pass by reference to `TemplateRef`.

Implementation reference:
- https://github.com/TEAMMATES/teammates/blob/master/src/web/app/components/toast/toast.component.html
- https://github.com/TEAMMATES/teammates/blob/master/src/web/app/components/toast/toast.component.ts
- https://github.com/TEAMMATES/teammates/blob/master/src/web/app/components/toast/toast.ts

Usage example:
https://github.com/TEAMMATES/teammates/blob/1bbf28caa2bef26f3bc148a61ed03608eac9d35b/src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.html#L47-L52
https://github.com/TEAMMATES/teammates/blob/1bbf28caa2bef26f3bc148a61ed03608eac9d35b/src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.ts#L21
https://github.com/TEAMMATES/teammates/blob/1bbf28caa2bef26f3bc148a61ed03608eac9d35b/src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.ts#L72

File: src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java [hunk 7]
```diff
@@ -210,7 +204,7 @@ public void resetInstructorGoogleId(InstructorAttributes instructor) {
         instructorRow.findElement(By.linkText(LINK_TEXT_RESET_GOOGLE_ID)).click();
         waitForPageToLoad();
 
-        resetGoogleIdModal.findElement(By.className("btn-warning")).click();
+        waitForConfirmationModalAndClickOk();
```
Comment (t-cheepeng) [inline src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java:207]: Possible to split this method into two? so `waitForConfirmationModal()` and `clickOkOnModal()`

File: src/web/app/components/simple-modal/simple-modal.component.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * A component to display contents of confirmation modals.
+ */
+@Component({
+  selector: 'tm-confirmation-modal',
+  templateUrl: './simple-modal.component.html',
+  styleUrls: ['./simple-modal.component.scss'],
+})
+export class SimpleModalComponent implements OnInit {
```
Comment (t-cheepeng) [inline src/web/app/components/simple-modal/simple-modal.component.ts:13]: 👍

File: src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java (hunk 7 above)
Comment (ccyccyccy) [reply src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java:207]: What's the benefit of doing that? There's a `waitForModalShown()` already, and almost all modal actions are doable by `waitForConfirmationModalAndClickOk` and `waitForConfirmationModalAndClickCancel()` anyway.

Comment (t-cheepeng) [reply src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java:207]: Yeah, just thought we could be clearer by separating the steps up further. But agree that modal actions are simple enough so let's leave it

File: src/web/services/simple-modal.service.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import { Injectable, TemplateRef } from '@angular/core';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { SimpleModalType } from '../app/components/simple-modal/simple-modal-type';
+import { SimpleModalComponent } from '../app/components/simple-modal/simple-modal.component';
+
+/**
+ * Optional paramters for modal.
+ */
+export interface SimpleModalOptions {
+  isInformationOnly?: boolean; // determines if there is 2 buttons for confirmation or only 1 button to close the modal
```
[the remark this answers is not in the document]
Comment (ccyccyccy) [reply src/web/services/simple-modal.service.ts:10]: There's a type of modal that only have 1 button, where the modal simply displays information with no need for confirmation.

image

File: src/web/app/pages-instructor/student-profile/more-info/more-info.component.ts [hunk 10]
```diff
@@ -14,16 +15,18 @@ export class MoreInfoComponent implements OnInit {
...
 
   /**
    * Open the more info modal.
    */
-  openModal(content: any): void {
-    this.ngbModal.open(content);
+  openMoreInfoModal(): void {
+    this.simpleModalService
+        .openInformationModal(`<strong>${this.studentName}<strong>\'s Profile - More Info`, SimpleModalType.NEUTRAL,
+            this.moreInfoText, { confirmMessage: 'Close' });
```
[the remark this answers is not in the document]
Comment (ccyccyccy) [reply src/web/app/pages-instructor/student-profile/more-info/more-info.component.ts:29]: Original:
image

After this PR:
image

Looks similar enough to me? The original simply wastes more space.

File: src/web/services/simple-modal.service.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/services/simple-modal.service.ts:10]: Yes, I am requesting to change the word "is" to "should be"

File: src/web/app/pages-instructor/student-profile/more-info/more-info.component.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-instructor/student-profile/more-info/more-info.component.ts:29]: @ccyccyccy you should check when `moreInfoText` is empty

[the remark this answers is not in the document]
Comment (ccyccyccy) [reply src/web/app/pages-instructor/student-profile/more-info/more-info.component.ts:29]: The option to open the modal shouldn't even show if `moreInfoText` is empty though. I could change the implementation to remove the redundancy.

File: src/web/app/components/question-types/question-edit-answer-form/contribution-question-edit-answer-form.component.ts [hunk 11]
```diff
@@ -44,10 +47,8 @@ export class ContributionQuestionEditAnswerFormComponent
     return points;
   }
 
-  /**
-   * Opens a modal.
-   */
-  openModal(modal: any): void {
-    this.modalService.open(modal);
+  openHelpModal(): void {
```
Comment (xpdavid) [inline src/web/app/components/question-types/question-edit-answer-form/contribution-question-edit-answer-form.component.ts:50]: You can make `equalShareHelp` as a argument here so you don't need to declare `@ViewChild('equalShareHelp')`

File: src/web/app/components/question-types/question-statistics/contribution-question-statistics/contribution-question-statistics.component.html [hunk 12]
```diff
@@ -60,57 +60,51 @@
...
-        <code>[Equal share]</code> because in both cases all members have done a similar share of work.
-      </li>
-      <li>
-        The sum of values in the team's view has been scaled to equal the sum of values in your view.
-        That way, you can make a direct comparison between your view and the team's view.
-        As this scaling is specific to you, the values you see in team's view may not match the values your team members see in their results.
-      </li>
-    </ul>
-  </div>
+<ng-template #contributionHelpModal>
```
Comment (xpdavid) [inline src/web/app/components/question-types/question-statistics/contribution-question-statistics/contribution-question-statistics.component.html:63]: Same here.

File: src/web/app/pages-help/instructor-help-page/instructor-help-questions-section/instructor-help-questions-section.component.ts [hunk 13]
```diff
@@ -126,18 +127,29 @@ export class InstructorHelpQuestionsSectionComponent extends InstructorHelpSecti
   };
 
   @Output() collapsePeerEvalTips: EventEmitter<any> = new EventEmitter();
+  @ViewChild('contributionTemplateBody') contributionTemplateBody!: TemplateRef<any>;
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-questions-section/instructor-help-questions-section.component.ts:130]: And in multiple cases

**Pull request merged**